filegenerator/file_generator.go

```diff
@@ -0,0 +1,393 @@
+				GeometricMeanData{NMean: nMean, AMean: aMean, BMean: bMean, MMean: mMean, Date: date})
+		}
+	}
+	fileName := deployRoot + "/geometric_means.json"
+	geometricMeansJSON, err := json.Marshal(&packageToGeometricMean)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = ioutil.WriteFile(fileName, geometricMeansJSON, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// GetGeometricMean returns the geometric mean of a vector.
+func GetGeometricMean(vector []float64) float64 {
+	epsilon := .0001
+	sum := 0.0
+	size := 0
+	/// Use law of logs to avoid overflows when multiplying entire vector.
+	for _, num := range vector {
+		if num > epsilon {
+			sum += math.Log(num)
+			size++
+		}
+	}
+	if size == 0 {
+		return 0
+	}
+	return math.Exp(sum / float64(size))
+}
+
+func getFormattedDate() string {
+	timeData := strings.Split(strings.Split(time.Now().String(), " ")[0], "-")
+	timeFormatted := timeData[2] + "-" + timeData[1] + "-" + timeData[0]
+	return timeFormatted
+}
+
+// This function creates all the necessary directories and subdirectories
+// for a sampleBench folder that mimics what the Marc folders look like
+func prepDirectories(dirs []string, date string) {
+	benchSamples := os.Getenv(benchSamplesEnv)
+	cmd := exec.Command("mkdir", date)
+	cmd.Dir = benchSamples
+	runWithStandardOutputs(cmd)
+	cmd = exec.Command("mkdir", "cockroach")
+	cmd.Dir = benchSamples + "/" + date
+	runWithStandardOutputs(cmd)
+	// Some of the directory strings are two levels deep, i.e. sql/parser.
+	// we need to iterate through the first level of directories in oreer to
+	// be able to create the second level.
+	// TODO (Will Haack) Handle N level of directories
+	for _, dir := range dirs {
+		matched, _ := regexp.MatchString("/", dir)
+		cockroachDir := benchSamples + "/" + date + "/" + "cockroach"
+		if matched {
+			topLevelDir := strings.Split(dir, "/")[0]
+			secondLevelDir := strings.Split(dir, "/")[1]
+			cmd := exec.Command("mkdir", topLevelDir)
+			cmd.Dir = cockroachDir
+			runWithStandardOutputs(cmd)
+			cmd = exec.Command("mkdir", secondLevelDir)
+			cmd.Dir = cockroachDir + "/" + topLevelDir
+			runWithStandardOutputs(cmd)
+		} else {
+			cmd := exec.Command("mkdir", dir)
+			cmd.Dir = cockroachDir
+			runWithStandardOutputs(cmd)
+		}
+	}
+}
+
+// GenerateBenchmarkResults runs the benchmark tests for today and saves the
+// results in the format that can be parsed by RenderHistoricalBenchMarkResults
+// TODO (Will Haack) Make sure that this runs on a recent branch.
+// TODO (Will Haack) Potentially remove this method if results are already being
+// generated and saved correctly.
+func GenerateBenchmarkResults(dirs []string) {
+	benchSamples := os.Getenv(benchSamplesEnv)
+	cockroachRoot := os.Getenv(cockroachRootEnv)
+	formattedDate := getFormattedDate()
+	prepDirectories(dirs, formattedDate)
+	for _, dirName := range dirs {
+		cmd := exec.Command("go", "test", "-benchmem", "-bench", ".", "-run", "XXX")
```

Comment:
Just a string that won't match, taken from http://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go